src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -194,58 +207,179 @@ public JsonWebKey lookupKey(
     {
         private static final int MAX_SCOPES = 48;
 
-        private final Map<String, Long> scopeBitsByName = new CopyOnWriteHashMap<>();
+        private final List<OAuthRealmInfo> realmInfos = new CopyOnWriteArrayList<>();
 
-        private final long realmId;
         private final String realmName;
 
-        private long nextScopeBit;
+        private int nextScopeBit = 0;
```

Comment:
Good decision to use narrower `int` type instead of `long`.
Remove explicit assignment to `0`, this is automatic by Java spec.